flask/src/test/groovy/net/corda/gradle/flask/FlaskPluginTest.groovy

```diff
@@ -0,0 +1,111 @@
+
+@CompileStatic
+class FlaskPluginTest {
+
+    private void installResource(String root, String resourceName, Path destination) {
+        Path outputFile = with {
+            Path realDestination
+            if (Files.isSymbolicLink(destination)) {
+                realDestination = destination.toRealPath()
+            } else {
+                realDestination = destination
+            }
+            realDestination = realDestination.resolve(resourceName)
+            if(!Files.exists(realDestination)) {
+                    Files.createDirectories(realDestination.parent)
+                    realDestination
+            } else if(Files.isDirectory(realDestination)) {
+                realDestination.resolve(resourceName.substring(1 + resourceName.lastIndexOf('/')))
+            }
+            else if(Files.isRegularFile(realDestination)) {
+                realDestination
+            } else throw new IllegalStateException("Path '${realDestination}' is neither a file nor a directory")
+        }
+        String resourcePath = root + '/' + resourceName
+        InputStream inputStream = (this.class.getResourceAsStream(resourcePath)
+                ?: this.class.classLoader.getResourceAsStream(resourcePath))
+        if(inputStream) {
+            inputStream.withStream {
+                Files.copy(it, outputFile, StandardCopyOption.REPLACE_EXISTING)
+            }
+        } else {
+            throw new FileNotFoundException(resourceName)
+        }
+    }
+
+    @TempDir
+    public Path testGradleHomeDir
+
+    @TempDir
+    public Path testProjectDir
+
+    Path buildFile
+
+    @BeforeEach
+    void setup() {
+        buildFile = testProjectDir.resolve("build.gradle")
+        installResource("testProject", "build.gradle", testProjectDir)
+        installResource("testProject", "settings.gradle", testProjectDir)
+        installResource("testProject","src/main/java/net/corda/gradle/flask/test/Main.java", testProjectDir)
+        installResource("testProject","src/main/java/net/corda/gradle/flask/test/AlternativeMain.java", testProjectDir)
+        installResource("testProject","src/flask/java/net/corda/gradle/flask/test/TestLauncher.java", testProjectDir)
+        installResource("testProject", "testAgent/build.gradle", testProjectDir)
+        installResource("testProject", "testAgent/src/main/java/net/corda/gradle/flask/test/agent/JavaAgent.java", testProjectDir)
+    }
+
+    GradleRunner getStandardGradleRunnerFor(String taskName) {
+        return GradleRunner.create()
+                .withDebug(true)
+                .withProjectDir(testProjectDir.toFile())
+                .withArguments(taskName, "-s", "--info", "-g", testGradleHomeDir.toString())
+                .withPluginClasspath()
+    }
+
+    @Test
+    void buildFlaskJar() {
+        GradleRunner runner = getStandardGradleRunnerFor("flaskJar")
+        BuildResult result = runner.build()
+    }
+
+    @Test
+    void runFlaskJar() {
```

Comment:
I think I should mention that while we will run Corda 5 on Java 11, the Corda Gradle plugins build still runs on Java 8 which does not support Java argument files. And the 5.x branch of our plugins needs to remain Java 8 compatible too. As a temporary fix, we can do:
```java
@EnabledForJreRange(min = JRE.JAVA_11)
@Test
void runFlaskJar() {
```
but Gradle 6.7 should allow us to add something like this to the test project's `build.gradle` file instead:
```
java {
    toolchain {
        languageVersion = JavaLanguageVersion.of(11)
    }
}
```

Reply:
oh, sure.. I was expecting version 6 of the Gradle plugin to switch to Java 11

Reply:
Gradle's [compatibility matrix](https://docs.gradle.org/current/userguide/compatibility.html) is here, which shows that Gradle still expects to be able to run on Java 8. I would prefer if our Gradle plugins could align with Gradle, if they possibly can. And our other plugins don't need anything specific to Java 11.

Reply:
well, that compatibility matrix says Gradle is fine with every Java version between 8 and 15 and we could build on java 11 compiling to java 8 bytecode to avoid breaking compatibility with Corda 4.x (but maybe we have already broken that in 6.x for several other reasons, I don't know). My point is that `runFlaskJar` is basically the most important of all unit tests and it would be good to have it running on CI.

Reply:
That's where the `toolchain` in the test project's `build.gradle` file comes in. I want this test running on Jenkins too.

Reply:
ok, for now I've simply annotated `net.corda.gradle.flask.FlaskPluginTest#runFlaskJarMainClassOverride` and `net.corda.gradle.flask.FlaskPluginTest#runFlaskJar` with `@EnabledForJreRange(min = JRE.JAVA_11)` to avoid breking tests on Java 8

Reply:
This is just temporary. Gradle 6.7's `toolchain` functionality will allow us to build every plugin with Java 8 but _test_ `flask` with Java 11:
```gradle
import static org.gradle.jvm.toolchain.JavaLanguageVersion.of

tasks.withType(Test).configureEach {
    javaLauncher = javaToolchains.launcherFor {
        languageVersion = of(11)
    }
}
```